server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -213,6 +221,13 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(formatter.pattern());
             out.writeString(timeZone.getId());
             out.writeVInt(resolution.ordinal());
+            if (out.getVersion().onOrAfter(Version.V_7_7_0) && out.getVersion().before(Version.V_8_0_0)) {
```

Comment:
similar comment as before `out.getVersion().onOrAfter(Version.V_7_7_0)` I don't think is strictly necessary. But nothing wrong with keeping it.